### PR TITLE
MGDAPI-3377 dont set target namespace in 3scale operatorGroup in mulitenant install

### PR DESCRIPTION
**Pull request by austincunningham**
# Issue link

https://issues.redhat.com/browse/MGDAPI-3377

# What

This PR leaves 3scale operator namespace scoped on sandbox upgrade using a cluster scoped 3scale operator, 

# Verification steps

- Deploy this catalogSource to install the base image
```yaml
apiVersion: operators.coreos.com/v1alpha1
kind: CatalogSource
metadata:
  name: rhmi-operators 
  namespace: openshift-marketplace
spec:
  sourceType: grpc
  image: quay.io/austincunningham/managed-api-service-index:1.18.2
```
- Run these make commands
```bash
INSTALLATION_TYPE=multitenant-managed-api make cluster/prepare/local
```
- create the following ClusterRole 
```yaml
kind: ClusterRole
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: sandboxClusterRole
rules:
  - verbs:
      - get
      - watch
      - list
    apiGroups:
      - '*'
    resources:
      - '*'
```
- and ClusterRoleBindings
```yaml
kind: RoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: sandboxRoleBinding
  namespace: sandbox-rhoam-operator
subjects:
  - kind: ServiceAccount
    name: rhmi-operator
    namespace: sandbox-rhoam-operator
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: sandboxClusterRole
```

- create the rhmi CR
```bash
INSTALLATION_TYPE=multitenant-managed-api USE_CLUSTER_STORAGE=true make deploy/integreatly-rhmi-cr.yml
```

- In the OperatorHub UI install rhoam in the `sandbox-rhoam-operator` namespace
- once the install is completed you can see the 3scale operator is namespaces scoped by going to operator/installed operators screen in the openshift UI
image
- change the CatalogSource to point at the upgrade image. `quay.io/austincunningham/managed-api-service-index:1.19.0`
- You will need to approve the upgrade for RHOAM
- Once the install is finish you will see in installed operators that 3scale is still namespaces scoped. 
image

### CODE REVIEW

**Comment by austincunningham** [comment]
we don't need ci/prow/multitenant-rhoam-e2e to pass currently broken

**Comment by den-rgb** [comment]
- Set up a cluster by following verification steps and installed rhoam under the sandbox-operator namespace.
- Upgraded rhoam to the newer image.
- 3scale operator was still namespace scoped.

File: pkg/resources/marketplace/manager.go [hunk 1]
```diff
@@ -88,14 +92,25 @@ func (m *Manager) InstallOperator(ctx context.Context, serverClient k8sclient.Cl
...
+
 	//TODO can maybe return this to a create only function after all clusters are updated in production
 	//err = serverClient.Create(ctx, og)
 	_, err = controllerutil.CreateOrUpdate(ctx, serverClient, og, func() error {
 		og.Namespace = t.Namespace
 		og.Name = OperatorGroupName
 		og.Labels = map[string]string{"integreatly": t.SubscriptionName}
 		if csv.Status.Phase == "Succeeded" && csv.Status.Reason == "InstallSucceeded" {
-			og.Spec.TargetNamespaces = operatorGroupNamespaces
+			if !integreatlyv1alpha1.IsRHOAMMultitenant(integreatlyv1alpha1.InstallationType(sandboxRhmiCR.Spec.Type)) {
```
Comment (MStokluska) [inline pkg/resources/marketplace/manager.go:111]: are we not going to have a nil pointer here at some point?
If the RHMI CR has not been found, we are sending in empty rhmi CR?

**Comment by austincunningham** [comment]
@briangallagher looks like this didn't merge last week , had to rebase this morning as the base branch has changed can you give it another `/lgtm` & `/approve`

**Comment by austincunningham** [comment]
@briangallagher putting this on hold looks like the e2e were broken by the last PR merged

**Comment by austincunningham** [comment]
@briangallagher looks like the e2e are passing now, No code changes , can you stick an `/lgtm` and `/approve` on this again

**Pull request merged**